model_package_core/tests/test_train.py

```diff
@@ -0,0 +1,36 @@
+import json
+from pathlib import Path
+
+from model_package_core.constants import (METADATA_FILENAME,
+                                          TRAIN_IMAGE_DATA_PATH,
+                                          TRAIN_LABEL_DATA_PATH,
+                                          WEIGHT_FILENAME)
+from model_package_core.pytorch_model import config
+from model_package_core.train import train
+
+
+def test_train_process(local_model_path):
```

Comment:
train is a very large function containing several other functions you've already unit tested. IMO its overkill to test the entire function. Pytorch will be an extremely well-tested library, so I'd focus purely on the pieces of code you wrote around it. 

For example, you could test that the `save_metadata` function is incrementing the version numbers as expected. 

If you think other pieces of `train` are important to test, can you extract them to a function and then test them alone without requiring the other `train` set up? Another way to think about this is, can you write tests that don't depend on pytorch being installed?